diagnostics/jaeger/grpc_tracer.go

```diff
@@ -0,0 +1,188 @@
+/*
+ * Copyright 2021 Layotto Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package jaeger
+
+import (
+	"context"
+	"os"
+	"time"
+
+	"github.com/opentracing/opentracing-go"
+	jaegerc "github.com/uber/jaeger-client-go"
+	"github.com/uber/jaeger-client-go/config"
+	"mosn.io/api"
+	"mosn.io/mosn/pkg/log"
+	"mosn.io/mosn/pkg/trace"
+	"mosn.io/mosn/pkg/trace/jaeger"
+	"mosn.io/mosn/pkg/types"
+
+	ltrace "mosn.io/layotto/components/trace"
+	"mosn.io/layotto/diagnostics/grpc"
+	"mosn.io/layotto/diagnostics/protocol"
+)
+
+const (
+	serviceName              = "layotto"
+	strategy                 = "strategy"
+	agentHost                = "agent_host"
+	collectorEndpoint        = "collector_endpoint"
+	defaultServiceName       = "layotto"
+	defaultJaegerAgentHost   = "127.0.0.1:6831"
+	jaegerAgentHostKey       = "TRACE"
+	appIDKey                 = "APP_ID"
+	defaultCollectorEndpoint = "http://127.0.0.1:14268/api/traces"
+	defaultStrategy          = "collector"
+)
+
+type grpcJaegerTracer struct {
+	tracer opentracing.Tracer
+}
+
+type grpcJaegerSpan struct {
+	*ltrace.Span
+	ctx        context.Context
+	trace      *grpcJaegerTracer
+	jaegerSpan opentracing.Span
+	spanCtx    jaegerc.SpanContext
+}
+
+func init() {
+	trace.RegisterTracerBuilder(jaeger.DriverName, protocol.Layotto, NewGrpcJaegerTracer)
```

Comment:
I think we can remove this `init` function since we already register it in the `main.go`.

We can also submit a new PR to delete the `init` func in skywalking tracer and `diagnostics/exporter_iml/stdout.go`.
Would like to hear your opinion
@LXPWing @wenxuwan @stulzq

Reply:
I can submit a PR to remove these init functions